cli/azd/pkg/tools/github/github.go

```diff
@@ -424,6 +444,46 @@ func (cli *Cli) GitHubActionsExists(ctx context.Context, repoSlug string) (bool,
 	return true, nil
 }
 
+func (cli *Cli) CreateEnvironmentIfNotExist(ctx context.Context, repoName string, envName string) error {
+	// Doc: https://docs.github.com/en/rest/deployments/environments?apiVersion=2022-11-28#create-or-update-an-environment
+	runArgs := cli.newRunArgs("api",
+		"-X", "PUT",
+		fmt.Sprintf("/repos/%s/environments/%s", repoName, envName),
+		"-H", "Accept: application/vnd.github+json",
+	)
+
+	res, err := cli.commandRunner.Run(ctx, runArgs)
+	if err != nil {
+		return err
+	}
+
+	if res.ExitCode != 0 {
+		return err
+	}
```

Comment:
minor: 1. `if res.ExitCode != 0 { return err }` seems unnecessary, and 2. it may be better to call `cli.run()`, notes below.

```suggestion
	res, err := cli.run(ctx, runArgs)
	if err != nil {
		return fmt.Errorf("creating new gh environment: %w", err)
	}
```


Notes:
1. On `if res.ExitCode != 0 { return err }`: a non-nil err is always returned if `ExitCode != 0`. Supposedly, if the err was nil, `if res.ExitCode != 0 { return err }` would end up returning `nil` anyways.
2. Using `cli.run()` seems to be the "right option" -- it wraps the error return for not logged in..

Reply:
I'll swap it out - using commandRunner directly wasn't really a choice on my part :)

Reply:
Done. I also wasn't using 'res', so I can just return the error directly-ish.